Proteomics/Protein/ProteoformLevelClassifier.cs

```diff
@@ -0,0 +1,223 @@
+            List<string> firstPTMsSortedByPTM = firstPTMsSortedByIndex.Select(x => x.ptm).OrderBy(x => x).ToList(); //sort ptms alphabetically
+            //check if there are unknown mass shifts
+            bool ptmsIdentified = !PtmsContainUnknownMassShifts(firstPTMsSortedByPTM);
+            bool ptmsLocalized = true; //assume these are localized unless we determine otherwise
+            //for every other sequence reported
+            for (int seqIndex = 1; seqIndex < sequences.Length; seqIndex++)
+            {
+                List<(int index, string ptm)> currentPTMsSortedByIndex = GetPTMs(sequences[seqIndex]); //get ptms from this sequence
+                List<string> currentPTMsSortedByPTM = currentPTMsSortedByIndex.Select(x => x.ptm).OrderBy(x => x).ToList(); //sort ptms alphabetically
+
+                //are number of PTMs the same?
+                if (firstPTMsSortedByIndex.Count == currentPTMsSortedByIndex.Count)
+                {
+                    //check localization (are indexes conserved?)
+                    for (int i = 0; i < firstPTMsSortedByIndex.Count; i++)
+                    {
+                        if (firstPTMsSortedByIndex[i].index != currentPTMsSortedByIndex[i].index)
+                        {
+                            ptmsLocalized = false;
+                            break;
+                        }
+                    }
+                    //check PTMs
+                    for (int i = 0; i < firstPTMsSortedByPTM.Count; i++)
+                    {
+                        if (!firstPTMsSortedByPTM[i].Equals(currentPTMsSortedByPTM[i]))
+                        {
+                            ptmsIdentified = false;
+                            break;
+                        }
+                    }
+                }
+                else
+                {
+                    ptmsIdentified = false;
+                    ptmsLocalized = false;
+                }
+            }
+
+
+            //determine gene ambiguity
+            bool geneIdentified = genes.Length == 1;
+
+
+            return GetProteoformClassification(ptmsLocalized, ptmsIdentified, sequenceIdentified, geneIdentified);
+        }
+
+        /// <summary>
+        /// Determine proteoform level between 1 (know everything) and 5 (only know the mass)
+        /// </summary>
+        /// <param name="ptmLocalized">Is the PTM localized?</param>
+        /// <param name="ptmIdentified">Do we know what the PTM is, or is it ambiguous (or an unknown mass shift?)</param>
+        /// <param name="sequenceIdentified">Do we know the proteoform sequence, or is it ambiguous?</param>
+        /// <param name="geneIdentified">Do we know which gene produced this proteoform?</param>
+        /// <returns></returns>
+        public static string GetProteoformClassification(bool ptmLocalized, bool ptmIdentified, bool sequenceIdentified, bool geneIdentified)
```

Comment:
Switch to private method?

Reply:
I think we should leave this one public so downstream users can still access it (some programs may produce different output than the fullSequenceString/geneString required for the ClassifyPrSM method. I'll change others to private, though.